lang/java/reef-io/src/main/java/org/apache/reef/io/network/group/impl/driver/GroupCommDriverImpl.java

```diff
@@ -140,22 +136,33 @@ public GroupCommDriverImpl(final ConfigurationSerializer confSerializer,
         .build())
         .build();
 
-    NameResolver nameResolver = null;
+    final NameResolver nameResolver;
     try {
       nameResolver = Tang.Factory.getTang().newInjector(nameResolverConf).getInstance(NameResolver.class);
     } catch (final InjectionException e) {
-      throw new RuntimeException(e);
+      throw new RuntimeException("Failed to instantiate NameResolver", e);
     }
 
-    this.netService = new NetworkService<>(idFac, 0, nameResolver,
-        new GroupCommunicationMessageCodec(), tpFactory,
-        new EventHandler<Message<GroupCommunicationMessage>>() {
-
-          @Override
-          public void onNext(final Message<GroupCommunicationMessage> msg) {
-            groupCommMessageStage.onNext(Utils.getGCM(msg));
-          }
-        }, new LoggingEventHandler<Exception>(), localAddressProvider);
+    try {
+      final Injector injector = TANG.newInjector();
```

Comment:
If we create a new injector and use so many bindVolatileParameter here, that doesn't make too much difference comparing to use new to create NetworkService. A proper solution is to use existing injector, and then the NetworkService would be automatically injectable.